### ISSUE TITLE
Researching basic projects

### ISSUE THREAD

**Issue by albfan** [albfan/miracast-windows-sink#1]
hmm, as I think, Windows should provide samples for his features: Let's start from here:

- https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/Projection
- https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/WiFiDirect

**Comment by albfan** [comment]
more examples (for wifi direct):

https://docs.microsoft.com/en-us/uwp/api/windows.devices.wifidirect

**Comment by albfan** [comment]
https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/WiFiDirect discover other wifi direct devices (connector) or answer to others trying to pair (advertiser). It exposes a service that allows both devices two send messages to each other.

Devices:

- android phone: huawey p10
- laptop: windows 10 laptop (afanjulport stands for Alberto Fanjul portatil)

This is the app finding my phone:

![image](https://user-images.githubusercontent.com/220968/45978928-cd0d0f00-c04d-11e8-932f-bf95e338e41d.png)

Or the app answering to a request from my phone to connect

![screenshot_20180924-230333](https://user-images.githubusercontent.com/220968/45979090-4278df80-c04e-11e8-9c7e-8960cc5aa3a9.jpg)

![image](https://user-images.githubusercontent.com/220968/45978982-f7f76300-c04d-11e8-9c52-f168624e1bc2.png)

Since service is a custom implementation I cannot make it work as my phone is not a windows 10 (so I should implement it on android) But anyone using both windows 10 phones should make it work out of the box

**Comment by albfan** [comment]
Second project https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/Projection

shows how to launch content to a know secondary display 

https://docs.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.projectionmanager

![image](https://user-images.githubusercontent.com/220968/45981559-b8cd1000-c055-11e8-9a56-81f5f116cafb.png)

Here the configuration of existing projects like https://github.com/albfan/miraclecast or https://github.com/intel/wds can help to configure the first example to expose a device by wifi direct that will be recognized as a secondary display (that is: implement miracast protocol)

About interaction with source, UIBC protocol allows sink device to send keyboard and mouse events to source

**Comment by fcgll520** [comment]
hi, do you have some progress about the windows implementation?

**Comment by albfan** [comment]
I need a Windows machine to do further test. No sure if my dual boot windows 10 is still valid. Will do a quick check soon.

**Comment by fcgll520** [comment]
i saw some softwares installed on windows, they use "WiFiDisplay.dll" to implement the miracast, 
do you have any idea about this library?

**Comment by albfan** [comment]
Windows come with support out of the box for Miracast (or WiFi display) so I think that's what that library is. Not sure why you ask here for a Windows DLL library, it it is for coding, Miracast is a protocol over WiFi direct, so you should be able to code it using this WiFi  direct example https://docs.microsoft.com/en-us/windows/iot-core/connect-your-device/setupwifidirect. Maybe there's any other Miracast example.

**Comment by fcgll520** [comment]
yes, it's a right direction, if i want to mirror an android's screen & audio to a PC by miracast protocol, not only windows devices.
do you have any idea?

**Comment by albfan** [comment]
A PC can mean anything (linux, BSD, windows) I guess you refer to windows. That should work out of the box on windows 10.

**Comment by fcgll520** [comment]
yes, the APIs of windows are not opened to the public, so it's hard to use.
I just know "WFDDisplaySinkStartEx", but i don't know what its params are.
